exec/cnex/array.c

```diff
@@ -70,6 +70,18 @@ void array_freeArray(Array *self)
     free(self);
 }
 
+void array_clearArray(Array *self)
+{
+    if (self != NULL && self->data != NULL) {
+        for (size_t i = 0; i < self->size; i++) {
+            cell_clearCell(&self->data[i]);
+        }
+        free(self->data);
+    }
+    self->data = NULL;
```

Comment:
Should this be inside the check for `self != NULL`? What if `self` actually is `NULL`?

Reply:
Good call!  Fixed.